backend/protocol_rpc/endpoints.py

```diff
@@ -531,6 +536,113 @@ def set_finality_window_time(consensus: ConsensusAlgorithm, time: int) -> None:
     consensus.set_finality_window_time(time)
 
 
+def get_chain_id() -> str:
+    return hex(61_999)
+
+
+def get_net_version() -> str:
+    return "61999"
+
+
+def get_block_number(transactions_processor: TransactionsProcessor) -> str:
+    transaction_count = transactions_processor.get_highest_timestamp()
+    return hex(transaction_count)
+
+
+def get_block_by_number(
+    transactions_processor: TransactionsProcessor, block_number: str, full_tx: bool
+) -> dict | None:
+    try:
+        block_number_int = int(block_number, 16)
+    except ValueError:
+        raise JSONRPCError(f"Invalid block number format: {block_number}")
+
+    block_details = transactions_processor.get_transactions_for_block(
+        block_number_int, include_full_tx=full_tx
+    )
+
+    if not block_details:
+        raise JSONRPCError(f"Block not found for number: {block_number}")
+
+    return block_details
+
+
+def get_gas_price() -> str:
+    gas_price_in_wei = 20 * 10**9
+    return hex(gas_price_in_wei)
+
+
+def get_transaction_receipt(
+    transactions_processor: TransactionsProcessor,
+    transaction_hash: str,
+) -> dict | None:
+
+    transaction = transactions_processor.get_transaction_by_hash(transaction_hash)
+
+    if not transaction:
+        return None
+
+    receipt = {
+        "transactionHash": transaction_hash,
+        "transactionIndex": hex(0),
+        "blockHash": transaction_hash,
+        "blockNumber": hex(transaction.get("block_number", 0)),
+        "from": transaction.get("from_address"),
+        "to": transaction.get("to_address") if transaction.get("to_address") else None,
+        "cumulativeGasUsed": hex(transaction.get("gas_used", 21000)),
+        "gasUsed": hex(transaction.get("gas_used", 21000)),
+        "contractAddress": (
+            transaction.get("contract_address")
+            if transaction.get("contract_address")
+            else None
+        ),
+        "logs": transaction.get("logs", []),
+        "logsBloom": "0x" + "00" * 256,
```

Comment:
will we fill this with something at some point?

Reply:
yes, this will be fixed in https://github.com/yeagerai/genlayer-studio/pull/755 when this data is going to be pulled from the hardhat network